src/app/Plans/components/Wizard/GeneralForm.tsx

```diff
@@ -1,77 +1,82 @@
+  const vmwareProviders = providersQuery.data?.vsphere || [];
+  const openshiftProviders = providersQuery.data?.openshift || [];
 
-  const sourceProvidersOptions = Object.values(sourceProviders).map((provider) => ({
+  if (providersQuery.isLoading) {
+    return <LoadingEmptyState />;
+  }
+  if (providersQuery.isError) {
+    return <Alert variant="danger" title="Error loading providers" />;
+  }
+
+  const sourceProvidersOptions = Object.values(vmwareProviders).map((provider) => ({
     toString: () => provider.name,
     value: provider,
   })) as OptionWithValue<IVMwareProvider>[];
 
-  const targetProvidersOptions = Object.values(targetProviders).map((provider) => ({
+  const targetProvidersOptions = Object.values(openshiftProviders).map((provider) => ({
     toString: () => provider.name,
     value: provider,
   })) as OptionWithValue<IOpenShiftProvider>[];
 
   return (
     <Form>
-      <Title headingLevel="h1" size="md">
+      <Title headingLevel="h2" size="md">
         Give your plan a name and a description
       </Title>
-      <FormGroup
+
+      <ValidatedTextInput
+        field={form.fields.planName}
         label="Plan name"
         isRequired
-        fieldId="planName"
-        helperTextInvalid="TODO"
-        validated="default" // TODO add state/validation/errors to this and other FormGroups
-      >
-        <TextInput id="planName" value={planName} type="text" onChange={setPlanName} />
-      </FormGroup>
+        fieldId="plan-name"
+      />
 
-      <FormGroup
+      <ValidatedTextInput
+        component={TextArea}
+        field={form.fields.planDescription}
         label="Plan description"
-        fieldId="planDescription"
-        helperTextInvalid="TODO"
-        validated="default" // TODO add state/validation/errors to this and other FormGroups
-      >
-        <TextArea
-          id="planDescription"
-          type="text"
-          value={planDescription}
-          onChange={setPlanDescription}
-        />
-      </FormGroup>
+        fieldId="plan-description"
+      />
 
-      <Title headingLevel="h3" size="md" /* className={styles.fieldGridTitle} */>
+      <Title headingLevel="h3" size="md">
         Select source and target providers
       </Title>
+
       <FormGroup
-        label="Source Provider"
+        label="Source provider"
         isRequired
-        fieldId="sourceProvider"
-        helperTextInvalid="TODO"
-        validated="default" // TODO add state/validation/errors to this and other FormGroups
+        fieldId="source-provider"
+        helperTextInvalid={form.fields.sourceProvider.error}
+        validated={form.fields.sourceProvider.isValid ? 'default' : 'error'}
       >
         <SimpleSelect
-          id="sourceProvider"
+          id="source-provider"
+          aria-label="Source provider"
           options={sourceProvidersOptions}
-          value={[sourceProvidersOptions.find((option) => option.value === sourceProvider)]}
+          value={[
```

Comment:
What do we do if there is no available provider to select?
![image](https://user-images.githubusercontent.com/11218376/95259708-9863ee80-07f5-11eb-83a8-e0348b390010.png)

Reply:
Hmm, that's a bug I was seeing before that I thought I fixed in this PR. Looking now.

Reply:
Pushed changes to prevent opening the wizard if there are insufficient providers.